ruby/test/ci/queue/redis_supervisor_test.rb

```diff
@@ -39,6 +39,18 @@ def test_wait_for_workers
     assert_equal true, workers_done
   end
 
+  def test_wait_for_workers_timeout
+    @supervisor = supervisor(timeout: 10, queue_init_timeout: 0.1)
```

Comment:
I'm not really happy with this test because we can't really distinguish between if we died because of the general timeout or all workers died. I could set the timeout a lot higher but still, the expectation of the test would still be the same. We could add some logging which we assert.

Reply:
Yeah, asserting logs sounds OK to me, especially since that's an error condition you'd want to log anyway.